src/hardware/clock.cpp

```diff
@@ -0,0 +1,36 @@
+#include <stm32h7xx_hal.h>
+
+/**
+ * @brief Stores the CPU clockrate.
+ *
+ * This is equivalent to `rcc_c_ck`, which is the same as `sys_ck` scaled by
+ * `D1CPRE`.
+ */
+uint32_t SystemCoreClock = 64000000;
+
+/**
+ * @brief Stores the Domain 2 AHB 1&2 clocks.
+ *
+ * This is eqiuvalent to `rcc_hclk`, which is the same as @ref SystemCoreClock
+ * scaled by `HPRE`.
+ */
+uint32_t SystemD2Clock = 64000000;
```

Comment:
I assume there's more to come later? This is just preliminary setup for the peripherals you needed?